frontend/apps/reader/modules/readerlink.lua

```diff
@@ -685,7 +719,7 @@ end
 
 function ReaderLink:getPreviousLocationPages()
     local previous_locations = {}
-    if #self.location_stack > 0 then
+    if self.location_stack_index > 0 or self.location_stack[self.location_stack_index+1] ~= nil then
```

Comment:
Same here. You'll be returning "previous location pages" stuff even when we are back at 0 ?!
I guess this is for Book map, and it feels it shouldn't show forward locations, if there is no way to distinguish them from previous locations...

Reply:
We can either have it show forward locations also, or only previous locations by breaking the `ipairs` loop on `location_stack_index`

Reply:
I guess this would feel more natural ? (if it works when _index is 0)
```diff
- for num, location in ipairs(self.location_stack) do
+ for num=1, location_stack_index do
+    location = self.location_stack[num]
```

Reply:
> We can either have it show forward locations also, or only previous locations by breaking

As for Book map, we could show previous forward locations with [Unicode white circled numbers](http://xahlee.info/comp/unicode_circled_numbers.html), but well, I don't think it would be really useful. Most people would get confused seeing in there stuff they are long back from.

Reply:
I will stick to only showing old ones